subs/pantry/src/Pantry/Types.hs

```diff
@@ -148,6 +155,13 @@ data Package = Package
   }
   deriving (Show, Eq)
 
+data PackageCabal = PCCabalFile !TreeEntry -- ^ TreeEntry of Cabal file
+                  | PCHpack
+                    !TreeEntry -- Original hpack file
+                    !TreeEntry -- Generated Cabal file
```

Comment:
Super-nit: If someone adds a field to this product and then adds an underscore to wherever it's being matched on, they won't get a type error.

Reply:
Fixed in the latest push.